### DISCUSSION TITLE
Add nginx reverse proxy example

### DISCUSSION THREAD

**Discussion by PaulGoldschmidt** [alfio-event/alf.io#1056] [Deployment]
This issue is a direct consequence of running alf.io in an docker environment (using docker-compose) and being stuck with the "development mode" notice, although the site access is no problem (see also #363 and #834).
Is there an example of a nginx reverse proxy? 

This is my nginx config for the reverse proxy right now (the site is of course served with SSL), maybe someone can spot the error - after finding a working configuration, I'll happily do a PR for the complete nginx example.

`location / {
proxy_pass http://127.0.0.1:8013;
proxy_set_header Host $host;
proxy_set_header X-Forwarded-Proto https;
proxy_set_header X-Forwarded-Host $host;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
}`

Kind regards,
Paul

**Comment by cbellone** [comment]
hi,
for reference here's a working configuration:

```
location / {
    proxy_pass http://localhost:8080;
    proxy_set_header  X-Real-IP $remote_addr;
    proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header  Host $http_host;
    proxy_set_header  X-Forwarded-Proto $scheme;
}
```

I'm converting this issue to a discussion, since it does not have any direct impact on alf.io's code

**Reply by PaulGoldschmidt** [reply]
Hey,
thanks a lot for the reply.
Interesting - I used almost the same config and even with the exact same config alf.io still starts in dev mode (with the profiles `profiles: requested , active dev, jdbc-session, spring-boot, stdout` given at startup). I doubt that the issue is related with Cloudflare as the choosen CDN, is there a setting I have to edit directly in the Docker container to deactivate the message?

Thanks for your help, really enjoying using Alf.io so far.
Best regards,
Paul

**Comment by cbellone** [comment]
Please double check the value of the
```
SPRING_PROFILES_ACTIVE
```
environment variable in the docker-compose descriptor. 
Default is `dev,jdbc-session`. If you want to run it in production you have to remove "dev", so you should have something like this:

```
SPRING_PROFILES_ACTIVE=jdbc-session
```

**Comment by PaulGoldschmidt** [comment]
That was it! Thanks for the help, really appreciate it! Totally missed the argument in the section of the docker-compose.

All the best,
Paul

**Comment by vb-03** [comment]
Hi
I'm running the site directly of the .war file, however when I'm out of dev mode I just get redirected to :8443 and the site threws me an error. I'm using the following the nginx config:


```
server {
        listen 80;
        listen [::]:80;

        #root /var/www/my_domain/html;
        #index index.html index.htm index.nginx-debian.html;
        #server_name my_domain www.my_domain;

        location / {
                try_files $uri $uri/ =404;
                proxy_pass http://localhost:8080;
                proxy_set_header  X-Real-IP $remote_addr;
                proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header  Host $http_host;
                proxy_set_header  X-Forwarded-Proto $scheme;
        }
}

```
Is there anything I'm doing wrong?

EDIT: Got it working with this virtual host config
For reference: #363 , #511 , #632 , #834 , #841  and https://gist.github.com/flexwie/a8572e8e7ef06ae844925ec857fa3b5b

```
server {
    listen      80;
    server_name my_domain;
    rewrite     ^   https://$server_name$request_uri? permanent;
}

server {
    listen 443 ssl;
    server_name my_domain;

    ssl_certificate /etc/letsencrypt/live/my_domain/cert.pem;
    ssl_certificate_key /etc/letsencrypt/live/my_domain/privkey.pem;

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_pass http://localhost:8080;
    }
}

```
